pkg/spiffs/fs/spiffs_fs.c

```diff
@@ -222,28 +222,31 @@ static int _open(vfs_file_t *filp, const char *name, int flags, mode_t mode, con
     (void) abs_path;
     DEBUG("spiffs: open: private_data = %p\n", filp->mp->private_data);
 
-    spiffs_flags s_flags = SPIFFS_O_RDONLY;
-    if (flags & O_APPEND) {
+    spiffs_flags s_flags = 0;
+    if ((flags & O_RDONLY) == O_RDONLY) {
```

Comment:
Sorry to comment once again on what I just told you to change. I realize that we will set SPIFFS_O_RDONLY on files even when they are opened with O_WRONLY.
We should change this to use O_ACCMODE as mask instead. O_ACCMODE is defined as the combined mask (O_WRONLY|O_RDONLY|O_RDWR)